File: Algorithm/LeetCode/daily/2024_09/02_2024_mid.ts
/**
 * @Time 2024/09/09 22:41
 * @Author  : https://github.com/hakusai22
 */

function maxConsecutiveAnswers(answerKey: string, k: number): number {
    let ans = 0, left = 0;
    const cnt = [0, 0];
    for (let right = 0; right < answerKey.length; right++) {
        cnt[answerKey[right].charCodeAt(0) >> 1 & 1]++;
        while (cnt[0] > k && cnt[1] > k) {
            cnt[answerKey[left++].charCodeAt(0) >> 1 & 1]--;
        }
        ans = Math.max(ans, right - left + 1);
    }
    return ans;
};

File: Algorithm/LeetCode/daily/2024_09/02_2024_mid.go
package _024_09

/*
    @Author  : https://github.com/hakusai22
    @Time    : 2024/09/09 22:41
    @题目     : https://leetcode.cn/problems/maximize-the-confusion-of-an-exam/description/?envType=daily-question&envId=2024-09-02
    @参考     : https://leetcode.cn/problems/maximize-the-confusion-of-an-exam/?envType=daily-question&envId=2024-09-02
    @时间复杂度: O(n)
    @空间复杂度:

 数据范围:

*/

func maxConsecutiveAnswers(answerKey string, k int) int {
	cnt := [2]int{}
	left := 0
	ans := 0
	for right, ch := range answerKey {
		cnt[ch>>1&1]++
		for cnt[0] > k && cnt[1] > k {
			cnt[answerKey[left]>>1&1]--
			left++
		}
		ans = max(ans, right-left+1)
	}
	return ans
}
